src/schema/geneInteraction.yaml

```diff
@@ -0,0 +1,467 @@
+      - rdf:Statement
+      - owl:Axiom
+      - biolink:association
+
+  GeneToGeneAssociation:
+    description: >-
+      abstract parent class for different kinds of gene-gene or gene product
+      to gene product relationships. Includes homology and interaction.
+    abstract: true
+    is_a: Association
+    defining_slots:
+      - subject
+      - object
+    slot_usage:
+      subject:
+        range: Gene
+        description: >-
+          the subject gene in the association. If the relation is symmetric,
+          subject vs object is arbitrary. We allow a gene product to stand
+          as a proxy for the gene or vice versa.
+      object:
+        range: Gene
+        description: >-
+          the object gene in the association. If the relation is symmetric,
+          subject vs object is arbitrary. We allow a gene product to stand
+          as a proxy for the gene or vice versa.
+    exact_mappings:
+      - biolink:gene_to_gene_association
+
+  GeneInteraction:
+    description: >-
+      An interaction between two genes or gene products; this may be a physical
+      molecular interaction between gene products (e.g. protein-protein interactions),
+      or may be a genetic interaction between genes (e.g. phenotypic suppression)
+    is_a: GeneToGeneAssociation
+    abstract: true
+    slots:
+      cross_references
+      data_provider
+      interaction_type
+      interactor_A_role
+      interactor_B_role
+      interactor_A_type
+      interactor_B_type
+    defining_slots:
+      - subject
+      - predicate
+      - object
+    id_prefixes:
+      - biogrid
+      - dip
+      - emdb
+      - empiar
+      - FB
+      - imex
+      - intact
+      - mint
+      - pdbe
+      - rcsb pdb
+      - WB
+      - wwpdb
+    slot_usage:
+      id:
+        description: >-
+          The unique primary identifier for the interaction. This should be the source
+          (curation) database identifier, or if that is not available then the aggregation
+          database identifier (e.g. IMEx)
+      cross_references:
+        description: >-
+          Additional identifier(s) of the interaction annotation other than the primary id.
+      subject:
+        range: Gene                   # is this declaration needed since it is already made in the parent class?
+        values_from:
+          - alliance
+      predicate:
+        subproperty_of: interacts_with
+        symmetric: true
+      relation:
+        values_from:
+          - RO
+          - mi
+      object:
+        range: Gene                   # is this declaration needed since it is already made in the parent class?
+        values_from:
+          - alliance
+      data_provider:
+        range: uriorcurie
+        description: >-
+          The database that curated the interaction. e.g. BioGRID
+        values_from:
+          - mi
+#        exact_mappings: # need to determine how to specify that the values should be an ontology class or descendant
```

Comment:
could use an enumeration as the range - do you want this under data_provider?

Reply:
Ideally, I would specify a particular ontology term (MI term in this case) under which (subclasses of which) would act as permissable values, otherwise I'll have to create quite a large list for an enumeration (and keep it updated). If we need an explicit enumeration for all possible values I can do that.

Reply:
I've commented out these two "values_from" instances.